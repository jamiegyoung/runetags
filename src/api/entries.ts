import { Entry } from '@/types';
import fs from 'fs';
// import LocatedItem from '@/api/locatedItem';

export const getTileData: () => Entry[] = () =>
  fs
    .readdirSync(`./src/tiles`, {
      encoding: `utf-8`,
      withFileTypes: true,
    })
    .filter((file) => file.isFile() && file.name.endsWith(`.json`))
    .map((file) => ({
      safeURI: file.name.replace(`.json`, ``),
      ...JSON.parse(
        fs.readFileSync(`./src/tiles/${file.name}`, { encoding: `utf-8` }),
      ),
    }))
    .map((tile) => ({
      ...tile,
      fullName: `${tile.name} ${tile.subcategory}`,
      fullAltName: `${tile.altName} ${tile.subcategory}`,
    }));

// const parseItemString = (itemString: string): LocatedItem[] =>
//   itemString.split(`,`).map((item) => {
//     const [id, location] = item.split(`:`);
//     return new LocatedItem({
//       id: Number(id),
//       location: Number(location),
//     });
//   });

// export const getTagData: () => Entry[] = () =>
//   fs
//     .readdirSync(`./src/tags`, {
//       encoding: `utf-8`,
//       withFileTypes: true,
//     })
//     .filter((file) => file.isFile() && file.name.endsWith(`.json`))
//     .map((file) => ({
//       safeURI: file.name.replace(`.json`, ``),
//       ...JSON.parse(
//         fs.readFileSync(`./src/tags/${file.name}`, { encoding: `utf-8` }),
//       ),
//     }))
//     .map((tag) => ({
//       ...tag,
//       items: parseItemString(tag.items) as LocatedItem[],
//       fullName: `${tag.name} ${tag.subcategory}`,
//       fullAltName: `${tag.altName} ${tag.subcategory}`,
//     }));
