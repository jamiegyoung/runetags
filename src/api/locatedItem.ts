import { Item, LocatedItemInterface } from '@/types';

export default class LocatedItem implements LocatedItemInterface {
  location: number;
  id: number;
  name: string;
  variants: Item[];
  iconURI: string;

  constructor(locatedItem: LocatedItemInterface) {
    this.id = locatedItem.id;
    this.name = locatedItem.name;
    this.variants = locatedItem.variants;
    this.iconURI = locatedItem.iconURI;
    this.location = locatedItem.location;
  }

  toString() {
    return `${this.id}:${this.location}`;
  }

  // if (this.layout) {
  //   return `banktaglayoutsplugin:${this.name},${this.items.reduce(
  //     (prev: string, curr) => {
  //       return `${prev},${curr.id}:${curr.location}`;
  //     },
  //     ``,
  //   )},banktag:${this.name},${this.iconItem.id},${this.items.reduce(
  //     (prev: string, curr) => {
  //       return `${prev},${curr.id}`;
  //     },
  //     ``,
  //   )}`;
  // }
  // return `${this.name},${this.iconItem.id},${this.items.reduce(
  //   (prev: string, curr) => {
  //     return `${prev},${curr.id}`;
  //   },
  //   ``,
  // )}`;
}
