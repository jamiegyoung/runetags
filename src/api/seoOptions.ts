import { NextSeoProps } from 'next-seo';

const nextSeoOptions: NextSeoProps = {
  defaultTitle: `RuneMarkers - A Collection of RuneLite Tile Markers`,
  description: `A collection of tile markers for RuneLite. Find and import tile markers for different Oldschool RuneScape activities.`,
  titleTemplate: `RuneMarkers - %s`,
  openGraph: {
    type: `website`,
    locale: `en_US`,
    url: `https://runemarkers.net`,
    title: `RuneMarkers - A Collection of RuneLite Tile Markers`,
    description: `A collection of tile markers for RuneLite. Find and import tile markers for different Oldschool RuneScape activities.`,
    siteName: `RuneMarkers`,
  },
  twitter: {
    handle: `@jamgyo`,
    site: `@runemarkers`,
    cardType: `summary_large_image`,
  },
};

export const defaultImages = [
  {
    url: `/logo1024-background.png`,
    width: 1024,
    height: 1024,
    alt: `RuneMarkers - A Collection of RuneLite Tile Markers`,
  },
  {
    url: `/logo512-background.png`,
    width: 512,
    height: 512,
    alt: `RuneMarkers - A Collection of RuneLite Tile Markers`,
  },
  {
    url: `/logo256-background.png`,
    width: 256,
    height: 256,
    alt: `RuneMarkers - A Collection of RuneLite Tile Markers`,
  },
  {
    url: `/logo128-background.png`,
    width: 128,
    height: 128,
    alt: `RuneMarkers - A Collection of RuneLite Tile Markers`,
  },
  {
    url: `/android-chrome-192x192.png`,
    width: 192,
    height: 192,
    alt: `RuneMarkers - A Collection of RuneLite Tile Markers`,
  },
  {
    url: `/android-chrome-512x512.png`,
    width: 512,
    height: 512,
    alt: `RuneMarkers - A Collection of RuneLite Tile Markers`,
  },
];

export default nextSeoOptions;
