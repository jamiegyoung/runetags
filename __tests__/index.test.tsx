import { getTileData } from '@/api/entries';
import '@testing-library/jest-dom';
import { act, fireEvent, render } from '@testing-library/react';
import Home, { getStaticProps } from '@/pages/index';

describe(`Home`, () => {
  it(`should search correctly`, () => {
    const data = getTileData();
    const { container, getByPlaceholderText } = render(
      <Home data={data} />,
    );
    const input = getByPlaceholderText(`type here to search`);
    act(() => {
      fireEvent.change(input, { target: { value: `abyssal sire` } });
    });
    expect(container).toMatchSnapshot();
  });

  it(`should generate static props`, () => {
    const tileData = getTileData();
    const { props } = getStaticProps();
    expect(props.data).toEqual(tileData);
  });
});
