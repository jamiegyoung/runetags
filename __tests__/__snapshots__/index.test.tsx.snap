// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should search correctly 1`] = `
<div>
  <div>
    <main
      class="main"
    >
      <div
        class="navbar"
      >
        <a
          href="./"
          style="background: transparent; color: rgb(245, 243, 245); text-decoration: none; display: flex;"
        >
          <header
            class="title"
            style="font-size: 2rem;"
          >
            R
            <span>
              une
            </span>
            <span
              class="titleAlt"
            >
              M
              <span>
                arkers
              </span>
            </span>
          </header>
        </a>
      </div>
      <div
        class="container"
      >
        <p
          style="padding: 0px; color: rgb(199, 199, 199); text-align: center; min-width: 280px; margin-bottom: 15px; max-width: 600px; width: 80%;"
        >
          Make sure the "Ground Markers" plugin is enabled on RuneLite with "Remember color per tile" enabled. Copy the tile markers to your clipboard, right-click the world map button in-game and select "import".
        </p>
        <div
          class="listOperationsContainer list"
        >
          <div
            style="display: flex; align-items: center; position: absolute; left: 50%; width: 100%; transform: translateX(-50%); justify-content: center;"
          >
            <input
              class="input "
              placeholder="type here to search"
              type="text"
              value="abyssal sire"
            />
            <button
              class="clearButton "
            >
              <svg
                height="24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6.4 19.45 4.55 17.6l5.6-5.6-5.6-5.6L6.4 4.55l5.6 5.6 5.6-5.6 1.85 1.85-5.6 5.6 5.6 5.6-1.85 1.85-5.6-5.6Z"
                />
              </svg>
            </button>
          </div>
          <button
            aria-label="Change layout"
            class="button"
            title="Change layout"
            type="button"
          >
            <div
              class="innerButton"
            >
              <div
                class="buttonSection list"
              />
              <div
                class="buttonSection list"
              />
              <div
                class="buttonSection list"
              />
              <div
                class="buttonSection list"
              />
              <div
                class="buttonSection list"
              />
              <div
                class="buttonSection list"
              />
            </div>
          </button>
        </div>
        <div
          class="innerListContainer"
        >
          <div
            class="card"
          >
            <a
              class="imageContainer"
              href="/abyssal%20sire"
            >
              <img
                alt="Abyssal Sire image"
                class="image"
                data-nimg="1"
                decoding="async"
                height="100"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Foldschool.runescape.wiki%2Fimages%2Fthumb%2FAbyssal_Sire.png%2F120px-Abyssal_Sire.png&w=256&q=75"
                srcset="/_next/image?url=https%3A%2F%2Foldschool.runescape.wiki%2Fimages%2Fthumb%2FAbyssal_Sire.png%2F120px-Abyssal_Sire.png&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Foldschool.runescape.wiki%2Fimages%2Fthumb%2FAbyssal_Sire.png%2F120px-Abyssal_Sire.png&w=256&q=75 2x"
                style="color: transparent; opacity: 0;"
                width="100"
              />
            </a>
            <div
              class="cardData"
            >
              <div
                class="entityInfo"
              >
                <a
                  class="name"
                  href="/abyssal%20sire"
                >
                  Abyssal Sire
                </a>
              </div>
              <div>
                <p
                  class="tileCount"
                >
                  6
                   tile markers
                </p>
                <div
                  class="tileInteraction"
                >
                  <button
                    class="button button"
                  >
                    Copy
                    <div
                      class="animBar "
                    />
                  </button>
                  <a
                    aria-label="Open Info Page for Abyssal Sire"
                    class="button"
                    href="/abyssal%20sire"
                  >
                    <svg
                      class="svg"
                      height="24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M11 17H13V11H11ZM12 9Q12.425 9 12.713 8.712Q13 8.425 13 8Q13 7.575 12.713 7.287Q12.425 7 12 7Q11.575 7 11.288 7.287Q11 7.575 11 8Q11 8.425 11.288 8.712Q11.575 9 12 9ZM12 22Q9.925 22 8.1 21.212Q6.275 20.425 4.925 19.075Q3.575 17.725 2.788 15.9Q2 14.075 2 12Q2 9.925 2.788 8.1Q3.575 6.275 4.925 4.925Q6.275 3.575 8.1 2.787Q9.925 2 12 2Q14.075 2 15.9 2.787Q17.725 3.575 19.075 4.925Q20.425 6.275 21.212 8.1Q22 9.925 22 12Q22 14.075 21.212 15.9Q20.425 17.725 19.075 19.075Q17.725 20.425 15.9 21.212Q14.075 22 12 22ZM12 12Q12 12 12 12Q12 12 12 12Q12 12 12 12Q12 12 12 12Q12 12 12 12Q12 12 12 12Q12 12 12 12Q12 12 12 12ZM12 20Q15.325 20 17.663 17.663Q20 15.325 20 12Q20 8.675 17.663 6.337Q15.325 4 12 4Q8.675 4 6.338 6.337Q4 8.675 4 12Q4 15.325 6.338 17.663Q8.675 20 12 20Z"
                      />
                    </svg>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <footer
        class="footer"
      >
        <div
          class="footerItem"
        >
          <p>
            Made by
            <a
              class="link"
              href="https://discord.com/users/167850724976361472"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="GitHub"
                data-nimg="1"
                decoding="async"
                height="17"
                loading="lazy"
                src="/_next/image?url=%2Fdiscord.png&w=48&q=75"
                srcset="/_next/image?url=%2Fdiscord.png&w=32&q=75 1x, /_next/image?url=%2Fdiscord.png&w=48&q=75 2x"
                style="color: transparent;"
                width="20"
              />
              <span
                class="linkText"
              >
                Jam#0001
              </span>
            </a>
          </p>
        </div>
        <div
          class="footerItem"
        >
          <a
            class="link"
            href="https://github.com/jamiegyoung/runemarkers"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="GitHub"
              data-nimg="1"
              decoding="async"
              height="20"
              loading="lazy"
              src="/_next/image?url=%2Fgithub.png&w=48&q=75"
              srcset="/_next/image?url=%2Fgithub.png&w=32&q=75 1x, /_next/image?url=%2Fgithub.png&w=48&q=75 2x"
              style="color: transparent;"
              width="20"
            />
            <span
              class="linkText"
            >
              contribute code / tiles
            </span>
          </a>
        </div>
      </footer>
    </main>
  </div>
</div>
`;
